src/lib/metric_ingest.py

```diff
@@ -215,25 +213,50 @@ async def fetch_metric(
             break
 
         for single_time_series in page['timeSeries']:
-            typed_value_key = extract_typed_value_key(single_time_series)
+            typed_value_key = _extract_typed_value_key(single_time_series)
             dimensions = create_dimensions(context, service_name, single_time_series, dt_dimensions_mapping, metric)
             entity_id = create_entity_id(service_name, service_dimensions, single_time_series)
 
             for point in single_time_series['points']:
-                line = convert_point_to_ingest_line(context, dimensions, metric, point, typed_value_key, entity_id)
+                line = _convert_point_to_ingest_line(context, dimensions, metric, point, typed_value_key, entity_id)
                 if line:
                     lines.append(line)
 
         next_page_token = page.get('nextPageToken', None)
         if next_page_token:
-            update_params(next_page_token, params)
+            _update_params(next_page_token, params)
         else:
             should_fetch = False
 
     return lines
 
 
-def update_params(next_page_token, params):
+def _set_aligner(metric_kind, value_type):
+    # Default, mainly for kind == DELTA (what we consider count)
+    aligner = 'ALIGN_SUM'
+
+    if metric_kind.lower().startswith('cumulative'):
+        aligner = 'ALIGN_DELTA'
+    elif metric_kind.lower().startswith('gauge'):
+        aligner = 'ALIGN_COUNT_TRUE' if value_type.lower() == 'bool' else 'ALIGN_MEAN'
+
+    return aligner
+
+
+def _set_reducer(metric_kind, value_type):
+    # Default, mainly for kind == DELTA (what we consider count)
+    reducer = 'REDUCE_SUM'
+
+    # Cannot be reduced
+    if metric_kind.lower().startswith('cumulative'):
+        reducer = 'REDUCE_NONE'
+    elif metric_kind.lower().startswith('gauge'):
+        reducer = 'REDUCE_COUNT_TRUE' if value_type.lower() == 'bool' else 'REDUCE_MEAN'
+
+    return reducer
```

Comment:
General comment about these 2 functions: would you consider creating dictionaries with the aligners and reducers? Then we will separate data from the functions, also you could assign keys like default_aligner, which often are better than comments. "Code never lies, comments sometimes do." 😄
Btw, if you do decide about using the dict, there is this [_MappingProxyType_](https://docs.python.org/3/library/types.html#types.MappingProxyType) which is immutable, which may be desired for such a use case.

Reply:
I agree with separating data from the functions, but I don't want to maintain keys that I won't use. As we have it now, I don't care for most of the possible values for the types, for instance. And I won't go half way, this is too simple for me to deserve it.

Reply:
Sure, valid points 😄